modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java

```diff
@@ -299,4 +303,60 @@ protected String formatWKT(final Formatter formatter) {
         formatter.setLinearUnit(linearUnit);
         return "PROJCS";
     }
+
+    @Override
+    public String formatPROJ(PROJFormatter formatter) {
+        final Ellipsoid ellipsoid = ((GeodeticDatum) datum).getEllipsoid();
+        @SuppressWarnings("unchecked") // Formatter.setLinearUnit(...) will do the check for us.
+        final Unit<Length> unit = (Unit) getUnit();
+        final Unit<Length> linearUnit = formatter.getLinearUnit();
+        final Unit<Angle> angularUnit = formatter.getAngularUnit();
+        final Unit<Length> axisUnit = ellipsoid.getAxisUnit();
+        formatter.setLinearUnit(unit);
+        formatter.setAngularUnit(
+                DefaultGeographicCRS.getAngularUnit(baseCRS.getCoordinateSystem()));
```

Comment:
Is see the geographic CRS already doing this... is it necessary to set the angular unit here as well?

Reply:
See my COMMENT1 for this

Reply:
Unit logic has been simplified.